### PR TITLE
Better support mongo replicas and variable retry params

### PR DESCRIPTION
**Pull request by andynemzek**

### CODE REVIEW

**Comment by andynemzek** [comment]
This must be merged before the analogous retrydb branch on fly/fly.  The changes here keep everything functionally the same.  It simply adds additional overridable configs so we can make use of them in our various environments.

File: api/config.py [hunk 1]
```diff
@@ -44,6 +44,8 @@
     },
     'persistent': {
         'db_uri': 'mongodb://localhost:9001/scitran',
+        'db_connect_timeout': 2000,
```
Comment (kofalt) [inline api/config.py]: Recommend making this higher, say `5000`

**Comment by kofalt** [comment]
LGTM after comment

**Comment by andynemzek** [comment]
@kofalt done.  Thanks!

**Pull request merged**